sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/Listeners/HttpRequestProcessor.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.WebJobs.Extensions.Storage.Blobs;
+using Microsoft.Azure.WebJobs.Extensions.Storage.Common;
+using Microsoft.Azure.WebJobs.Host.Blobs.Listeners;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Queues.Listeners;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Blobs
+{
+    internal class HttpRequestProcessor: IHttpRequestProcessor
+    {
+        private readonly SharedQueueWatcher _sharedQueueWatcher;
+        private readonly IHostIdProvider _hostIdProvider;
+        private readonly QueueServiceClientProvider _queueServiceClientProvider;
+        private readonly ILogger _logger;
+        private BlobTriggerQueueWriter _blobTriggerQueueWriter;
+
+        public HttpRequestProcessor(
+            SharedQueueWatcher sharedQueueWatcher,
+            IHostIdProvider hostIdProvider,
+            QueueServiceClientProvider queueServiceClientProvider,
+            ILoggerFactory loggerFactory)
+        {
+            _sharedQueueWatcher = sharedQueueWatcher;
+            _hostIdProvider = hostIdProvider;
+            _queueServiceClientProvider = queueServiceClientProvider;
+            _logger = loggerFactory.CreateLogger<HttpRequestProcessor>();
+        }
+
+        public void RegisterHttpEnpoint(ExtensionConfigContext context)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            Uri url = context.GetWebhookHandler();
+#pragma warning restore CS0618 // Type or member is obsolete
+            _logger.LogInformation($"registered http endpoint = {url?.GetLeftPart(UriPartial.Path)}");
+        }
+
+        public async Task<HttpResponseMessage> ProcessHttpRequestAsync(HttpRequestMessage req, CancellationToken cancellationToken)
+        {
+            var functionId = HttpUtility.ParseQueryString(req.RequestUri.Query)["functionName"];
```

Comment:
This seems to be a partial(?) copy of the event grid implementation.

Reply:
Do we need to include sources in this case?

Reply:
I wonder if this should be an extension for an extension. I.e. someone would need to pull extra package like `Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.EventGrid` in order to add extra functionality that spans across blobs and eventgrid. Some kind of plugin architecture.

Reply:
This code is different for blob and event grid. For event grid we check if function with the name exists, for blob we will check this latter.